front/pages/api/enrichment/company.ts

```diff
@@ -457,11 +453,18 @@ export default async function handler(
 
   const encodedEmail = encodeURIComponent(email);
 
-  const isExistingUser = await checkExistingUser(email);
-  if (isExistingUser) {
+  const regionAffinityResult = await checkUserRegionAffinity({
```

Comment:
If the goal is to check if the user exists (show login or signup), rather use fetchUsersFromWorkOSWithEmails . This one will try to check existing invites or domain (if email_verified true) to select the region, but you may have affinity without having an existing account. And you actually don't care about the region, you can anyway stick to `/api/workos/login?loginHint=${encodedEmail}`. Whatever region the user has, they will be redirected after login.